### PR TITLE
Episode Timeline Editing & Pain Location Validation Fixes

### PR DESCRIPTION
**Pull request by vfilby**
## Summary

This PR introduces comprehensive episode timeline editing capabilities and fixes critical pain location validation issues. Users can now edit episode end times, reopen episodes, and manage pain locations without validation errors.

### 🎯 Episode End Timeline Editing
- **Long press functionality** on "Episode Ended" timeline events
- **Edit end time** using existing date/time picker interface  
- **Reopen episodes** by removing end time (makes episode ongoing again)
- **iOS ActionSheet & Android Alert** with consistent UX patterns

### 🔧 Dedicated Reopen Episode Button
- **Edit screen integration** - Reopen button next to end time picker
- **Confirmation dialogs** prevent accidental reopens
- **Proper navigation** back to previous screen after success
- **Consistent styling** with danger color for destructive action

### 🏪 Enhanced Episode Store
- **Dedicated reopenEpisode function** with proper business logic
- **Current episode management** - Reopened episodes become active
- **Cache invalidation** for consistent state updates
- **Error handling** with user-friendly toast messages

### 🩹 Pain Location Validation Error
- **Fixed schema validation** that prevented removing all pain locations
- **Repository update handling** for undefined values using 'in' operator  
- **Test coverage expansion** for edge cases and validation scenarios
- **Backward compatibility** maintained for existing episodes

### 🧹 UI/UX Improvements  
- **Removed redundant Pain Locations card** from episode details
- **Timeline-centric approach** for all episode data management
- **Cleaner interface** with single source of truth for pain locations
- **Consistent interaction patterns** across all timeline events

### Database Layer
- Fixed `episodeRepository.update()` to handle `{ endTime: undefined }` correctly
- Updated validation logic to handle null/undefined end times properly
- Enhanced test coverage for reopen functionality

### State Management  
- Added `reopenEpisode` function to episode store
- Improved cache management for episode state updates
- Better error handling and user feedback

### UI Components
- Enhanced `EpisodeDetailScreen` with long press timeline editing
- Updated `NewEpisodeScreen` with reopen button integration  
- Removed duplicate pain location display components

## Testing
- ✅ All existing tests pass
- ✅ New test cases for reopen functionality
- ✅ Pain location validation edge cases covered
- ✅ Repository update handling validated

## User Impact
- **Improved workflow** for episode management
- **Fixed blocking validation errors** for pain location updates
- **Multiple reopen options** (timeline long press + edit screen button)
- **Cleaner UI** with reduced information duplication
- **Enhanced accessibility** with proper labels and hints

Addresses user feedback on episode editing limitations and pain location validation issues.

### CODE REVIEW

File: app/src/screens/EpisodeDetailScreen.tsx [hunk 1]
```diff
@@ -944,6 +934,94 @@ export default function EpisodeDetailScreen({ route, navigation }: Props) {
     }
   };
 
+  const handleEpisodeEndLongPress = () => {
+    if (!episode) return;
+
+    if (Platform.OS === 'ios') {
```
Comment (vfilby) [inline app/src/screens/EpisodeDetailScreen.tsx]: Perhaps a future task, but would it be a good idea to wrap the pattern so we don't end up with giant if iOS / else android actionsheet chunks scattered in the code base.
